### ISSUE TITLE
swiper.slidePrev() and  swiper.slideNext() it doesn't work without restarting the browser

### ISSUE THREAD

**Issue by wapewape** [twinssbc/Ionic2-Calendar#486]
i friend, I need a help
I have my calendar working but I have a problem
When you start the calendar, the buttons for changing the week, or going backwards or forwards, do not work.
I am printing on the console
const swiper = document.querySelector ('. swiper-container') ['swiper'];
console.log ('BACK swiper', swiper);
swiper.slidePrev ();
console.log ('swiper.slideNext ();', swiper.slidePrev ());
  and this are the results
![sin](https://user-images.githubusercontent.com/1838797/72266477-30cd9980-35fd-11ea-9ec9-26219177e3d9.png)

but if I restart the browser (google chrome or firefox) the option to change the week works correctly.
![despues](https://user-images.githubusercontent.com/1838797/72266561-5fe40b00-35fd-11ea-8b0a-3a0117e90fb5.png)

Do you have any idea how I can solve this?
  as you know in an app I can not execute the action of refreshing the entire page

**Comment by twinssbc** [comment]
@wapewape It seems your console print will also trigger the swiping? Could you remove it and only let your button control the swiping? Also it seems the code you provided doesn't match your screenshot. Could you provide the exact code you tested? And how do you trigger the swiping in the buttons?

**Comment by wapewape** [comment]
my friend explained to you
In the first image I am printing on the console by pressing the button to change the slide
console.log ('swiper.slideNext ();', swiper.slidePrev ()); and returns true but does not change the slide

on that same page when reloading the page without touching the code, it works correctly and returns false by changing the slide correctly

**Comment by twinssbc** [comment]
@wapewape I'm still confused about your code 

```
const swiper = document.querySelector ('. swiper-container') ['swiper'];
console.log ('BACK swiper', swiper);
swiper.slidePrev ();
console.log ('swiper.slideNext ();', swiper.slidePrev ());
```

If seems you called swiper.slidePrev twice? Also why do you need to trigger swiper.slidePrev() in the console.log? Do you realize it has side effect, not just printing some output? Your output contains NEXT, but I don't see how your code could print that out.

**Comment by wapewape** [comment]
I shared with you an small video explain my problem
https://drive.google.com/file/d/1uWZA1PjHjmL9zeQTQ4qjzy2kIXwPcRQl/view?usp=sharing
Please see

**Comment by twinssbc** [comment]
@wapewape Thanks for the video, especially with the music! I notice that you have another sliders in the page and before reproducing the problem, you also slide second couple times. Maybe it could be the problem.
Could you try below steps?
1. Could you try also slide the second slide after the refresh, then slide the calendar to see if the problem occurs?
2. Could you also try just slide the calendar the first time to see if the problem occurs?
3. Could you remove the second slide see if sliding the calendar works?

**Comment by wapewape** [comment]
Hello Friend
  I already checked what you mentioned and I don't get the problem
You could help me to check the code through remote assistance. I could pay you for this job. thanks

**Comment by twinssbc** [comment]
@wapewape Did you also check step 3? If you remove the second slide, you no longer meet this issue?
May I know which time zone are you in?
If it's not convenient to do remote assistance, is it possible you could send me your code? You could just leave the code that can reproduce the issue.

**Comment by wapewape** [comment]
time zone Isla de Pascua (GMT-5)

don't worry about the time, I can connect 24/7
I can also send you the code, you can leave me an email
thanks

**Comment by twinssbc** [comment]
@wapewape You could send code to the email address in my profile.

**Comment by jcmendes9898** [comment]
Sometimes i get the same problem! I need to restart the page in order to work. If you find a solution please share it.

**Comment by twinssbc** [comment]
@wapewape @jcmendes9898 I did some investigation and found probably it's because the document.querySelector way is directly manipulating the DOM element. But sometimes the DOM element is not rendered completely in time. I added two instance method slidePrev and slideNext which leverages the official exposed ion-slides method, so no longer doing some DOM level manipulation. The change is in the latest version 0.5.8 and how to use these two methods are also added in README. Feel free to give a try.
